File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        dp = [False] * n
        dp[0] = True
        for i in range(n):
            if dp[i] == False:
                return False
            if nums[i] + i >= n:
                return True
            for j in range(1, nums[i] + 1):
                dp[i+j] = True
            # print(dp)
        return dp[n-1]

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(); // Min PQ for kth Largest
        for (int i = 0; i < nums.length; i++) {
            pq.offer(nums[i]);
            if (pq.size() > k) {
                pq.poll();
            }
        }
        return pq.poll();
    }
}
